### PR TITLE
[3747] Document courses endpoint by year

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/Zth5QQPE/3747-document-api-endpoint-api-public-v1-recruitmentcycles-year-courses
- Document `/api/public/v1/recruitment_cycles/:year/courses`

### Changes proposed in this pull request

- Document the public courses endpoint scoped by year
- Move the existing courses endpoint scoped under a `providers` module
- Update `api_spec.json` to reflect current state of yaml

### Guidance to review

- Run the middleman docs, see the new endpoint documentation for courses by year
- Discuss whether it's worth making the endpoint snake_case rather than kebab-case to be consistent with other endpoints

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Document API endpoint - /api/public/v1/recruitment_cycles/:year/courses](https://trello.com/c/Zth5QQPE/3747-document-api-endpoint-api-public-v1-recruitmentcycles-year-courses)

**Comment by asmega** [comment]
So If we do it this way it means we'll have to manage 2 courses controller depending on the level of nesting

**Comment by darokel** [comment]
> So If we do it this way it means we'll have to maintain 2 courses controllers. One for year and one for provider

I think that's fine as they will potentially start to deal with different concerns. Otherwise we'd have to mix in logic to handle different constraints (year, provider etc) in one controller/action.

File: spec/api/courses_spec.rb [hunk 1]
```diff
@@ -46,50 +40,13 @@
                 example: { page: 2, per_page: 10 },
                 description: "Pagination options to navigate through the collection."
 
-      response "200", "The collection of courses." do
+      response "200", "The collection of courses for the specified year." do
```
Comment (defong) [inline spec/api/courses_spec.rb:43]:
```suggestion
      response "200", "The collection of courses." do
```
based on 
https://api2.qa.publish-teacher-training-courses.service.gov.uk/api-reference.html#recruitment_cycles-year-providers-get

File: spec/api/courses_spec.rb [hunk 2]
```diff
@@ -1,23 +1,17 @@
 require "swagger_helper"
 
 describe "API" do
-  path "/recruitment_cycles/{year}/providers/{provider_code}/courses" do
-    get "Returns the courses for the specified provider." do
-      operationId :public_api_v1_provider_courses
-      tags "course"
+  path "/recruitment_cycles/{year}/courses" do
+    get "Returns the courses for the specified year." do
```
Comment (defong) [inline spec/api/courses_spec.rb:5]:
```suggestion
    get "Returns the courses for the specified recruitment cycle." do
```

based on 
https://api2.qa.publish-teacher-training-courses.service.gov.uk/api-reference.html#recruitment_cycles-year-providers-get

**Pull request merged**